### PR TITLE
ddl: Use the correct error

### PR DESCRIPTION
**Pull request by zimulala**
1. Use the correct error
2. It is unreliable to check whether the error is equal to `grpc.ErrClientConnClosing`.  So I update this logic.

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @hicqu @disksing @tiancaiamao

**Comment by hicqu** [comment]
Perfect. LGTM.

File: ddl/syncer.go [hunk 1]
```diff
@@ -184,29 +183,36 @@ func (s *schemaVersionSyncer) RemoveSelfVersionPath() error {
 	return errors.Trace(err)
 }
 
-func isContextFinished(err error) bool {
-	if terror.ErrorEqual(err, goctx.Canceled) ||
-		terror.ErrorEqual(err, goctx.DeadlineExceeded) {
+func isCtxFinishedInETCD(err error) bool {
```
Comment (disksing) [inline ddl/syncer.go]: Should be `isCtxFinishedInEtcd`, etcd is not abbreviation.

File: ddl/owner_manager.go [hunk 2]
```diff
@@ -117,17 +116,17 @@ func newSession(ctx goctx.Context, flag string, etcdCli *clientv3.Client, retryC
...
 			break
 		}
 		log.Warnf("[ddl] %s failed to new session, err %v", flag, err)
-		if isContextFinished(err) || terror.ErrorEqual(err, grpc.ErrClientConnClosing) {
-			break
-		}
 		time.Sleep(200 * time.Millisecond)
-		continue
 	}
 	return etcdSession, errors.Trace(err)
```
Comment (tiancaiamao) [inline ddl/owner_manager.go]: `return etcdSession, nil`

File: ddl/syncer.go [hunk 3]
```diff
@@ -184,29 +183,36 @@ func (s *schemaVersionSyncer) RemoveSelfVersionPath() error {
 	return errors.Trace(err)
 }
 
-func isContextFinished(err error) bool {
-	if terror.ErrorEqual(err, goctx.Canceled) ||
-		terror.ErrorEqual(err, goctx.DeadlineExceeded) {
```
Comment (tiancaiamao) [inline ddl/syncer.go]: I'm afraid the error code may be codes.Unknown sometimes, even it's canceled.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by disksing** [comment]
LGTM.

**Pull request merged**